internal/controller/dscinitialization/dscinitialization_controller.go

```diff
@@ -564,3 +576,69 @@ func (r *DSCInitializationReconciler) watchHWProfileCRDResource(ctx context.Cont
 
 	return []reconcile.Request{{NamespacedName: types.NamespacedName{Name: instanceList.Items[0].Name}}}
 }
+
+// CreateGatewayConfig creates a default GatewayConfig if it doesn't exist.
+// Parameters:
+//   - ctx: context for the operation
+//   - instance: DSCInitialization instance
+//
+// Returns:
+//   - error: nil on success, error if GatewayConfig creation fails
+func (r *DSCInitializationReconciler) CreateGatewayConfig(ctx context.Context, instance *dsciv2.DSCInitialization) error {
```

Comment:
Is it not good to keep the gateway config creation and update logic inside `gateway_support.go`? Becuase the `dscinitialization_controller.go` file size keeps increasing.